src/middleware/builtin.ts

```diff
@@ -207,28 +207,41 @@ export function matchConstraints(
  */
 export function matchMessage(pattern: string | RegExp): Middleware<SlackEventMiddlewareArgs<'message'>> {
   return async ({ message, context, next }) => {
```

Comment:
please correct me if i'm wrong, but if we simply used the `event` argument instead of the `message` argument (a event of type message would have both in the listener arguments, and `message` is just an alias of `event`), then it seems we could just reuse the `matchMessage()` middleware for both use cases. personally, i think that would be better than creating a second built-in middleware that does _almost_ the exact same thing.

Reply:
@aoberoi I think you're right.

The reason I separated into two functions is because I supposed it was intentional that `message` was used instead of `event` (c.f. @shaydewael 's commit https://github.com/slackapi/bolt-js/commit/c9ac664ce76a3be15ebe4412d640b1d5bccff909 ) 

But I think `message` is just an alias of `event` too, so I updated this PR in the way `matchMessage()` took `event` so that it could handle `MessageEvent` and `AppMentionEvent`.

What do you think, @aoberoi ?